### PR TITLE
Enhancment/114 move to redux

### PR DESCRIPTION
**Pull request by lily-law**
This should mostly work. I need to tweak users. I've yet to update profileCard to redux (currently users can add selves as connections). 
And I still need to move search to redux. I'm thinking of changing from wrapping search around other page content to directing to the search page when a search is performed and back to page it was called from when closed, this should be easier now using redux.  
Edit: and I need to merge in the latest commits in development

### CODE REVIEW

**Comment by davideastmond** [comment]
> This should mostly work. I need to tweak users. I've yet to update profileCard to redux (currently users can add selves as connections).
> And I still need to move search to redux. I'm thinking of changing from wrapping search around other page content to directing to the search page when a search is performed and back to page it was called from when closed, this should be easier now using redux.
> Edit: and I need to merge in the latest commits in development

Are you going to address the points above in this same PR?

**Comment by lily-law** [comment]
> > This should mostly work. I need to tweak users. I've yet to update profileCard to redux (currently users can add selves as connections).
> > And I still need to move search to redux. I'm thinking of changing from wrapping search around other page content to directing to the search page when a search is performed and back to page it was called from when closed, this should be easier now using redux.
> > Edit: and I need to merge in the latest commits in development
> 
> Are you going to address the points above in this same PR?

Yes 💯, just putting it as a draft now so there are not too many changes to review when finished.
Maybe a bit late 😂 there are quite a lot already

**Review by davideastmond — APPROVED** [review]

File: app/src/pages/home/homeSlice.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+      // })
+      // .addCase(deleteThreadAsync.fulfilled, (state, action) => {
+      //   stateStatus.idle(state);
+      //   const newThreadsState = {...state.threads};
+      //   delete newThreadsState[action.payload.id];
+      //   state.threads = newThreadsState;
+      // })
+      // .addCase(deleteThreadAsync.rejected, (state) => {
+      //   stateStatus.error(state, "unable to delete thread");
+      // })
```
Comment (davideastmond) [inline app/src/pages/home/homeSlice.ts:187]: Should we remove this?

File: app/src/pages/home/homeSlice.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+      // .addCase(updateThreadCommentAsync.fulfilled, (state, action) => {
+      //   stateStatus.idle(state);
+      //   state.threads = {
+      //     ...state.threads,
+      //   //  [action.payload.id]: action.payload,
+      //   };
+      // })
+      // .addCase(updateThreadCommentAsync.rejected, (state) => {
+      //   stateStatus.error(state, "");
+      // })
```
Comment (davideastmond) [inline app/src/pages/home/homeSlice.ts:256]: Remove?

File: app/src/pages/profile/profileSlice.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  "profile/addConnection",
+  async ({
+    connectionId,
+    isTeamMate,
+  }: {
+    connectionId: string;
+    isTeamMate: boolean;
+  }) => {
+    const response = await addConnection({ connectionId, isTeamMate });
+    return response;
```
Comment (davideastmond) [inline app/src/pages/profile/profileSlice.ts:47]:
```suggestion
    return await addConnection({ connectionId, isTeamMate });
```

File: app/src/pages/profile/profileSlice.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  }
+);
+
+export const removeConnectionAsync = createAsyncThunk(
+  "profile/removeConnection",
+  async ({ connectionId }: { connectionId: string }) => {
+    const response = await removeConnection({ connectionId });
+    return response;
```
Comment (davideastmond) [inline app/src/pages/profile/profileSlice.ts:55]:
```suggestion
   return await removeConnection({ connectionId });
```

File: app/src/services/thread/thread.ts [hunk 5]
```diff
@@ -1,190 +1,178 @@
...
-        "Sorry, we're unable to remove your reaction at this time, please try again later"
-      );
+  const res = await axios({
+    method: "delete",
+    url: `/api/threads/${threadId}/likes`,
+    data: { threadLikeId },
+  });
+  if (res?.data?.updatedThread) {
+    const processedThreadData = await processThread(res.data.updatedThread);
+    return processedThreadData;
```
Comment (davideastmond) [inline app/src/services/thread/thread.ts:80]:
```suggestion
    return await processThread(res.data.updatedThread);
```

Comment (davideastmond) [inline app/src/services/thread/thread.ts:80]:
```suggestion
    return await processThread(res.data.updatedThread);
```

File: app/src/services/thread/thread.ts [hunk 6]
```diff
@@ -1,190 +1,178 @@
...
-        "Sorry, we're unable to add your post at this time, please try again later"
-      );
+  const res = await axios({
+    method: "post",
+    url: `/api/threads/${threadId}/comments`,
+    data,
+  });
+  if (res?.data?.updatedThread) {
+    const processedThreadData = await processThread(res.data.updatedThread);
+    return processedThreadData;
```
Comment (davideastmond) [inline app/src/services/thread/thread.ts:100]:
```suggestion
   return await processThread(res.data.updatedThread);
```
Are we going to be doing more processing with this variable?

File: app/src/services/user/user.ts [hunk 7]
```diff
@@ -1,89 +1,82 @@
...
+const updateUser = async ({ data }: { data: IUserPatchRequest }) => {
+  const req = await axios({
+    method: "patch",
+    url: `/api/users/me`,
+    data,
+  });
+  if (req.status === 200) {
+    return data;
+  } else {
+    throw req.statusText;
```
Comment (davideastmond) [inline app/src/services/user/user.ts]: We're throwing an error object here. Should we not do
```suggestion
    throw new Error(req.statusText);
``` ?

File: app/src/utils/cookie.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+function getCookie(name: string) {
+  const match = document.cookie.match(
+    RegExp("(?:^|;\\s*)" + name + "=([^;]*)")
+  );
+  return match ? match[1] : null;
+}
+
+// copied from https://www.w3schools.com/js/js_cookies.asp
```
Comment (davideastmond) [inline app/src/utils/cookie.ts:8]: 😎

**Review by davideastmond — COMMENTED** [review]
Looking good. I had a few comments and suggestions.
Also did a bit of a test - noticed a few quirks, but not sure if they'll be resolved when the API portion gets merged?

I think once this is done and merged, we can have a testing session and then create any tickets for any bugs that might arise?

File: app/src/pages/home/homeSlice.ts (hunk 1 above)
Comment (lily-law) [reply app/src/pages/home/homeSlice.ts:187]: Let's leave them for now if you've no strong objections. The implementation for these is what it is so I don't think they're speculating (not far off anyway) on what they'll be when uncommented. They can be uncommented now without issue but they're easier to spot commented.

File: app/src/services/thread/thread.ts (hunk 6 above)
Comment (lily-law) [reply app/src/services/thread/thread.ts:100]: I prefer to assign to something on these, when debugging I'll often want to console.log the awaited data.

**Comment by lily-law** [comment]
> Looking good. I had a few comments and suggestions.
> 
> Also did a bit of a test - noticed a few quirks, but not sure if they'll be resolved when the API portion gets merged?
> 
> 
> 
> I think once this is done and merged, we can have a testing session and then create any tickets for any bugs that might arise?

🙏 
Maybe when we voice chat today or tomorrow we could use a few mins just to go over the quirks, I'm not sure I'm aware of them yet.

**Pull request merged**